components/automate-ui/src/app/pages/service-groups/service-groups.component.ts

```diff
@@ -211,40 +210,34 @@ export class ServiceGroupsComponent implements OnInit, OnDestroy {
       withLatestFrom(this.route.queryParamMap),
       takeUntil(this.isDestroyed)
     )
-    .subscribe(([serviceGroups, queryParams]) => {
+    .subscribe(([serviceGroups]) => {
       if (serviceGroups.length > 0) {
-        // if we do not have an sgId parameter, pick the first sgId from the
-        // service group list and load it:
-        const sgId = queryParams.get('sgId');
-        if ( !sgId ) {
-          this.router.navigate(
-            [],
-            { queryParams: { sgId: serviceGroups[0]['id'] }, queryParamsHandling: 'merge' }
-          );
+        // if we do not have a selected service group, pick the first sgId from
+        // the service group list
+        if ( !this.selectedServiceGroupId ) {
+          this.selectedServiceGroupId = serviceGroups[0]['id'];
         }
 
         // if the selected service group is not visible on the page, then pick
-        // the first service group from the list and navigate to it. this lets
-        // us maintain a service group selection regardless of other navigation
-        // as long as the service group is still in the list.
-        const selectedSGVisibleOnPage = serviceGroups.find(sg => sg.id === sgId);
+        // the first service group from the list. this lets us maintain a
+        // service group selection regardless of other navigation as long as
+        // the service group is still in the list.
+        const selectedSGVisibleOnPage = serviceGroups.find(sg => {
+          return sg.id === this.selectedServiceGroupId;
+        });
```

Comment:
Could make slightly more readable/concise:
```
        const selectedSGVisibleOnPage =
          serviceGroups.find(sg => sg.id === this.selectedServiceGroupId);
 
```

Reply:
Yeah, I like that better, too but the line length linter doesn't like it 🙃